driver-core/src/main/com/mongodb/internal/dns/DnsResolver.java

```diff
@@ -41,28 +45,51 @@
     // 0 5 5060 example.com.
     // The priority and weight are ignored, and we just concatenate the host (after removing the ending '.') and port with a
     // ':' in between, as expected by ServerAddress
-    public static List<String> resolveHostFromSrvRecords(final String host) {
+    // It's required that the srvHost has at least three parts (e.g. foo.bar.baz) and that all of the resolved hosts have a parent
+    // domain equal to the domain of the srvHost.
+    public static List<String> resolveHostFromSrvRecords(final String srvHost) {
+        if (srvHost.split("\\.").length < 3) {
+            throw new IllegalArgumentException(format("An SRV host name '%s' was provided that does not contain at least three parts",
```

Comment:
Can we clarify this error message? Or at least name the three parts it should have - something like:

> "An SRV host name '%s' was provided that does not contain at least three parts. It must contain a hostname, domain name and tld."

Reply:
Done